apps/messages/lib.js

```diff
@@ -57,15 +57,23 @@ exports.pushMessage = function(event) {
   // otherwise load messages/show widget
   var loadMessages = Bangle.CLOCK || event.important;
   // first, buzz
-  var quiet = (require('Storage').readJSON('setting.json',1)||{}).quiet;
-  if (!quiet && loadMessages && global.WIDGETS && WIDGETS.messages)
+  var quiet       = (require('Storage').readJSON('setting.json',1)||{}).quiet;
+  var unlockWatch = (require('Storage').readJSON('setting.json',1)||{}).unlockWatch;
```

Comment:
You're reading setting.json here but 'unlockWatch' seems to be written to `messages.settings.json` ?